lib/pure/collections/deques.nim

```diff
@@ -75,64 +93,164 @@ template xBoundsCheck(deq, i) =
     if unlikely(i >= deq.count):  # x < deq.low is taken care by the Natural parameter
       raise newException(IndexError,
                          "Out of bounds: " & $i & " > " & $(deq.count - 1))
+    if unlikely(i < 0):  # when used with BackwardsIndex
```

Comment:
@dom96 You recently introduced accessing elements via `BackwardsIndex`, but if you pass illegal index, it would throw `RangeError` — that is inconsistent with `IndexError` which is raised when you pass illegal (not-backwards) index.

So I added this check, and I've put `xBoundsCheck(deq, deq.len - int(i))` in those procedures, so now they also throw `Indexerror`.

Thoughts?

Reply:
This is not required, you cannot use negative numbers for the builtin ``[]`` and it would raise the same exception.

Reply:
Without this part something like `a[^999]` raises `RangeError` (while `a[999]` raises `IndexError`).

Reply:
Hmm, must be caused by the ``Natural``. Very well then,